packages/router/src/instruction-resolver.ts

```diff
@@ -182,4 +127,41 @@ export class InstructionResolver {
     }
     return strings.join(this.separators.sibling);
   }
+
+  private _parseViewportInstruction(instruction: string): ViewportInstruction {
+    let component, viewport, parameters, scope;
+    const [componentPart, rest] = instruction.split(this.separators.viewport);
+    if (rest === undefined) {
+      [component, ...parameters] = componentPart.split(this.separators.parameters);
+      if (component.endsWith(this.separators.ownsScope)) {
+        scope = true;
+        component = component.slice(0, -1);
```

Comment:
```suggestion
        component = component.slice(0, -this.separators.ownsScope.length);
```
This makes it a bit more obvious what's going on here and also makes it resilient to potentially having more than one character as a separator (which doesn't appear to be forbidden anywhere)